plugins/woocommerce/src/Internal/Admin/EmailPreview.php

```diff
@@ -197,4 +181,23 @@ private function get_email() {
 		$email  = $emails['WC_Email_Customer_Processing_Order'];
 		return $email;
 	}
+
+	/**
+	 * Set up filters for email preview.
+	 */
+	private function set_up_filters() {
+		// Always show shipping address in the preview email.
+		add_filter( 'woocommerce_order_needs_shipping_address', '__return_true' );
+		// Email templates fetch product from the database to show additional information, which are not
+		// saved in WC_Order_Item_Product. This filter enables fetching that data also in email preview.
+		add_filter( 'woocommerce_order_item_product', array( $this, 'get_dummy_product_when_not_set' ), 10, 1 );
+	}
+
+	/**
+	 * Clean up filters after email preview.
+	 */
+	private function clean_up_filters() {
+		remove_filter( 'woocommerce_order_needs_shipping_address', '__return_true' );
```

Comment:
This could remove a filter added by someone else, because it uses a stock WP function name. Maybe it's better to create a function in this class and attach it, e.g. `add_filter( 'woocommerce_order_needs_shipping_address', array( $this, 'enable_shipping_address' ))`, so the function-to-be-removed will be unambiguous.

Reply:
Good catch! Addressed in ae46cb0.